test/regression_tests/thermal_bubble_regression_test.jl

```diff
@@ -11,7 +11,9 @@ function run_thermal_bubble_regression_test(arch, grid_type)
     end
 
     closure = IsotropicDiffusivity(ν=4e-2, κ=4e-2)
-    model = NonhydrostaticModel(architecture=arch, grid=grid, closure=closure, coriolis=FPlane(f=1e-4))
+    model = NonhydrostaticModel(architecture=arch, grid=grid, closure=closure, coriolis=FPlane(f=1e-4),
+                                buoyancy=Buoyancy(model=SeawaterBuoyancy()), tracers=(:T, :S),
```

Comment:
It's not super important, but can we use the "concise" form `buoyancy=SeawaterBuoyancy()` where applicable? Fewer characters and easier to read might help make test maintenance easier.

Reply:
I thought we were gonna try to move towards the more verbose way (`Buoyancy(model=Model())`) instead of allowing the two different ways of constructing it to be valid. (In the same spirit of enforcing `GradientBoundaryCondition`, instead of `BoundaryCondition(Gradient, ...)`, or however we used to do it.)

Is that not the case?

Reply:
Well, I prefer the concise form so I hope we can keep it:

1. `HydrostaticFreeSurfaceModel` cannot support gravity aligned with anything other than `z` (and we want similar APIs for both hydrostatic and non hydrostatic models)
2. Most nonhydrostatic use cases appear to be ok with gravity aligned with `z`

So the more verbose constructor creates boilerplate in a lot of scripts, and is only required for a small number of problems. I also don't think it adds clarity to the constructors, but others may feel differently... ?

The boundary condition issue is a little different --- there is a popular package for autodifferentiatpion and machine learning called `Flux`, so exporting the name `Flux` from Oceananigans makes it hard to use the two packages together. The "prefixed" constructors avoid this name conflict and are similarly verbose (and arguably more literal sine we are more likely to _say_ `GradientBoundaryCondition` than `BoundaryCondition`, `Gradient`) so we decided to use those instead.